### PR TITLE
Add dosomething_rogue module

### LINKED ISSUE #6958: Rogue settings menu
#### User Story

As an admin I want to configure rogue url and api keys on the phoenix DoSomething settings page.

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Adds a `dosomething_rogue` module to hold all rogue related features. 

Adds a global configuration form so we can configure api settings.

image
#### How should this be reviewed?

enable to the new `dosomething_rogue` module. go to `/admin/config/services/rogue` and you should be able to edit the settings there.
#### Relevant tickets

Fixes #6958

### CODE REVIEW

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @file
+ * Code for the dosomething_rogue feature.
+ */
+
+define('ROGUE_API_URL', variable_get('dosomething_rogue_url', 'https://qa-rogue.dosomething.org'));
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: set this default to `roge.dev` or whatever the local uri is, so we don't all connect to the qa instance locally

Comment (sbsmith86) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: ah good point!

Comment (sbsmith86) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: @angaither updated!

**Comment by angaither** [comment]
📷 ➡️

**Pull request merged**